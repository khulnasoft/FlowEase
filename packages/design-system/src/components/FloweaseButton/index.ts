import FloweaseButton from './Button.vue';

export default FloweaseButton;
