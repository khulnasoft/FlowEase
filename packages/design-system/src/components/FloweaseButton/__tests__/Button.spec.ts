import { render } from '@testing-library/vue';
import FloweaseButton from '../Button.vue';

const slots = {
	default: 'Button',
};
const stubs = ['flowease-spinner', 'flowease-icon'];

describe('components', () => {
	describe('FloweaseButton', () => {
		it('should render correctly', () => {
			const wrapper = render(FloweaseButton, {
				slots,
				global: {
					stubs,
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		describe('props', () => {
			describe('loading', () => {
				it('should render loading spinner', () => {
					const wrapper = render(FloweaseButton, {
						props: {
							loading: true,
						},
						slots,
						global: {
							stubs,
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});
			});

			describe('icon', () => {
				it('should render icon button', () => {
					const wrapper = render(FloweaseButton, {
						props: {
							icon: 'plus-circle',
						},
						slots,
						global: {
							stubs,
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});
			});

			describe('square', () => {
				it('should render square button', () => {
					const wrapper = render(FloweaseButton, {
						props: {
							square: true,
							label: '48',
						},
						global: {
							stubs,
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});
			});
		});
	});
});
