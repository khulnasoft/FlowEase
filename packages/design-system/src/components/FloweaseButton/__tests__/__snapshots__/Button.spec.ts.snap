// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > FloweaseButton > props > icon > should render icon button 1`] = `"<button class="button button primary medium withIcon" aria-live="polite"><span class="icon"><flowease-icon-stub icon="plus-circle" size="medium" spin="false"></flowease-icon-stub></span><span>Button</span></button>"`;

exports[`components > FloweaseButton > props > loading > should render loading spinner 1`] = `"<button class="button button primary medium loading withIcon" disabled="" aria-busy="true" aria-live="polite"><span class="icon"><flowease-spinner-stub size="medium" type="dots"></flowease-spinner-stub></span><span>Button</span></button>"`;

exports[`components > FloweaseButton > props > square > should render square button 1`] = `
"<button class="button button primary medium square" aria-live="polite">
  <!--v-if--><span>48</span>
</button>"
`;

exports[`components > FloweaseButton > should render correctly 1`] = `
"<button class="button button primary medium" aria-live="polite">
  <!--v-if--><span>Button</span>
</button>"
`;
